pysmt/solvers/z3.py

```diff
@@ -491,3 +492,52 @@ def eliminate_quantifiers(self, formula):
                                           expression=res)
 
         return pysmt_res
+
+
+class Z3Interpolator(Interpolator):
+
+    LOGICS = [QF_AUFLIA, QF_UFLRA]
```

Comment:
Currently, PySMT does not support arrays, so this should be QF_UFLIA. (At least, we did not included non-representable logics in other solvers / quantelims)